src/models/Observation.js

```diff
@@ -159,17 +152,19 @@ class Observation extends Realm.Object {
       user
     };
     if ( !existingObs ) {
-      localObs._created_at = new Date( );
+      localObs._created_at = new Date( localObs.created_at );
+      if ( isNaN( localObs._created_at ) ) {
+        localObs._created_at = new Date( );
+      }
     }
     return localObs;
   }
 
   static async saveLocalObservationForUpload( obs, realm ) {
     // make sure local observations have user details for ObsDetail
-    const id = await getUserId( );
-    const user = id && realm.objectForPrimaryKey( "User", Number( id ) );
-    if ( user ) {
-      obs.user = user;
+    const currentUser = realm.objects( "User" ).filtered( "signedIn == true" )[0];
```

Comment:
We have so many different ways to check for a current user...

Reply:
Agree, and I'm not making it better in this PR, but maybe the subject of a future issue?